### PR TITLE
Feature reserve form

### PR DESCRIPTION
**Pull request by MoyasiGinko**
In this milestone,

- Created a new branch
- Added the reservations form
- - From the details page
- - From the navbar
- Updated styling for all the components.

### CODE REVIEW

File: src/redux/reducers/resereveSlice.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+const createReservation = createAsyncThunk(
+  'reservation/createReservation',
+  async (reservationData, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(url, reservationData, { headers });
+      return response.data;
+    } catch (error) {
+      // Log the error for debugging purposes
+      console.error('Error:', error);
```
Comment (lincoln1883) [inline src/redux/reducers/resereveSlice.js:27]: please rewrite.

File: src/redux/reducers/rservationSlice.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+const fetchReservations = createAsyncThunk(
+  'reservations/fetchReservations',
+  async () => {
+    try {
+      const response = await axios.get(url, { headers });
+      return response.data;
+    } catch (error) {
+      console.error('Error:', error);
```
Comment (lincoln1883) [inline src/redux/reducers/rservationSlice.js:26]: Please rewrite.

File: src/redux/reducers/resereveSlice.js (hunk 1 above)
Comment (lincoln1883) [inline src/redux/reducers/resereveSlice.js:27]: same here.

File: src/redux/reducers/rservationSlice.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+const cancelReservation = createAsyncThunk(
+  'reservations/cancelReservation',
+  async (reservationId) => {
+    try {
+      const response = await axios.delete(`${url}${reservationId}`, {
+        headers,
+      });
+      return response.data;
+    } catch (error) {
+      console.error('Error:', error);
```
Comment (lincoln1883) [inline src/redux/reducers/rservationSlice.js:42]: Same here all there caused linter error after I fixed the merge conflict.

File: src/redux/reducers/resereveSlice.js (hunk 1 above)
Comment (ibugithub) [reply src/redux/reducers/resereveSlice.js:27]: @lincoln1883 OK, let me check and rewrite to handle the specific errors to show to the users.

**Review by lincoln1883 — APPROVED** [review]
Hi @MoyasiGinko @ibugithub , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights

- Implemented reservation feature ✔️
- Add styles ✔️
- Used Git flow ✔️
- Fix linter errors ✔️
- Fixed conflicts ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

**Pull request merged**